Client/U2FExtensions.swift

```diff
@@ -132,27 +136,30 @@ class U2FExtensions: NSObject {
             observeKeyStateUpdates = true
         }
         
-        popup = AlertPopupView(image: #imageLiteral(resourceName: "browser_lock_popup"), title: Strings.touchKeyTitle, message: Strings.touchKeyMessage)
+        touchKeyPopup = AlertPopupView(image: #imageLiteral(resourceName: "browser_lock_popup"), title: Strings.keyTitle, message: Strings.touchKeyMessage)
+        insertKeyPopup = AlertPopupView(image: #imageLiteral(resourceName: "browser_lock_popup"), title: Strings.keyTitle, message: Strings.insertKeyMessage)
+        pinVerificationPopup = AlertPopupView(image: #imageLiteral(resourceName: "browser_lock_popup"), title: Strings.pinLabel, message: Strings.pinTitle, inputType: .default, secureInput: true, inputPlaceholder: Strings.pinPlaceholder)
         super.init()
         
-        popup.addButton(title: Strings.touchKeyCancel) { [weak self] in
+        touchKeyPopup.addButton(title: Strings.keyCancel) { [weak self] in
             guard let self = self else {
                 return .flyDown
             }
-            let handle = self.currentHandle
-            switch self.currentMessageType {
-                case .FIDO2Create:
-                    self.sendFIDO2RegistrationError(handle: handle)
-                case .FIDO2Get:
-                    self.sendFIDO2AuthenticationError(handle: handle)
-                case.FIDORegister:
-                    self.sendFIDORegistrationError(handle: handle, requestId: self.requestId[handle] ?? -1, errorCode: U2FErrorCodes.other_error)
-                case .FIDOSign:
-                    self.sendFIDOAuthenticationError(handle: handle, requestId: self.requestId[handle] ?? -1, errorCode: U2FErrorCodes.other_error)
-                case .FIDOLowLevel, .None:
-                    break
+            return self.handleCancelButton()
+        }
+        
+        insertKeyPopup.addButton(title: Strings.keyCancel) { [weak self] in
+            guard let self = self else {
```

Comment:
These blocks are identical so should be split up:

```
let popupButtonClosure: () -> PopupViewDismissType = {
    // ... all this code
}

[touchKeyPopup, insertKeyPopup, pinVerificationPopup].forEach {
    $0.addButton(title: Strings.keyCancel, tapped: popupButtonClosure)
}
```

Reply:
looks much better, will update. Thanks!